### ISSUE TITLE
Implement ANN402

### ISSUE THREAD

**Issue by janosh** [charliermarsh/ruff#4460]
Flag and auto-fix [`flake8-annotations`](https://github.com/sco1/flake8-annotations#opinionated-warnings) `ANN402` type comments.

```diff
- nums = [] # type: List[float]
+ nums: list[float] = []  # if target=py3.10+ or future annotations imported

+ from typing import List
+ nums: List[float] = []  # else
```

**Comment by JonathanPlasse** [comment]
It is already flagged by `PYI033`.
Only the auto-fix remain to be implemented.

**Comment by janosh** [comment]
Could we add an alias from ANN402 to PYI033? or a link from one to the other in the docs?

**Comment by janosh** [comment]
@JonathanPlasse I just had a look at the docs. Actually `PYI033` only applies to stub files. It doesn't flag type comments in regular python files.

**Comment by calumy** [comment]
Red #4166 which makes a start on enabling this rule for .py files.

**Comment by p1-dta** [comment]
Hi, just noticed the missing 402 were missing from flake8-annotations, is there any plans to implement this in the near future?
